tree/ntuple/v7/inc/ROOT/RFieldVisitor.hxx

```diff
@@ -61,16 +61,16 @@ public:
    virtual void VisitFloatField(const RField<float> &field) { VisitField(field); }
    virtual void VisitByteField(const RField<std::byte> &field) { VisitField(field); }
    virtual void VisitCharField(const RField<char> &field) { VisitField(field); }
-   virtual void VisitInt8Field(const RField<std::int8_t> &field) { VisitField(field); }
-   virtual void VisitInt16Field(const RField<std::int16_t> &field) { VisitField(field); }
-   virtual void VisitInt32Field(const RField<std::int32_t> &field) { VisitField(field); }
-   virtual void VisitInt64Field(const RField<std::int64_t> &field) { VisitField(field); }
+   virtual void VisitInt8Field(const RIntegralField<std::int8_t> &field) { VisitField(field); }
```

Comment:
Shouldn't we pass an `RField<integer type>` here and for the others? `RIntegralField` is an implementation details, isn't it?

Reply:
So I agreed yesterday evening, but when looking into it I remembered why I decided to pass `RIntegralField`: the solution proposed in this PR builds on a surjective, but not injective function that maps any integral type (the `T` in `RField<T>`) onto one of the (fewer) fixed width integer types (the `T` in `RIntegralField`, mapped via `RIntegralTypeMap`). We *could* continue to pass `RField`s, but we wouldn't know which of `RField<long>`, `RField<long long>` and `RField<std::int64_t>` to implement in the interface (Linux maps `typedef long int64_t;` while macOS does `typedef long long int64_t;`).

In practical terms it would also not be nice to implement because `AcceptVisitor` is currently a part of `RIntegralField`, where `*this` doesn't have type `RField<T>`. We would either have to add dynamic casting in there, or move `AcceptVisitor` to the templated `RField<T>`. That in turn would again require specializations because the name of the `Visit` function depends on the type...